src/queue/handleRequests.ts

```diff
@@ -39,40 +41,89 @@ export async function handleRepositoriesByQueue(
     console.log('Handle task', task);
 
     if (!task) {
-      return Promise.reject('');
+      return Promise.reject('Empty task');
     }
 
-    let repository = {} as Repository;
-    if (repositoryArray.length) {
-      const [currentRepository, ...restRepositories] = repositoryArray;
+    const getRepositoriesTask = () => {
+      currentPage++;
+
+      const isLastPage =
+        currentPage !== 1 &&
+        currentPage === Math.round(repositoryLength / MAX_REPO_PER_PAGE);
+
+      return getRepositoriesRequest(language, currentPage).then(
+        (response): TaskResult => {
+          const remainingRateLimit = response.headers[RATE_LIMIT_HEADER];
+
+          if (Number(remainingRateLimit) === 0) {
+            queue.clear();
```

Comment:
Extend `TaskRepository` to data about error. Clear error in onSucceed callback if error was provided